Backpack/Calendar/Classes/BPKCalendarSelectionConfiguration.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2021 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKCalendarSelectionConfiguration : NSObject
+
+/**
+ * Creates a `BPKCalendarSelectionConfiguration` with the specific classes.
+ *
+ * @param rangeStyleSelection Whether the selection configuration is a range style.
+ * @param allowsMultipleSelection Whether the selection configuration allows multiple simultaneous selections.
+ * @return `BPKCalendarSelectionConfiguration` instance.
+ */
+- (instancetype)initWithRangeStyleSelection:(BOOL)rangeStyleSelection allowMultipleSelection:(BOOL)allowsMultipleSelection;
+
+/**
+ * Whether the selection configuration is a range style (ie start date and end date with 0 or more dates between).
+ */
+@property(readonly)BOOL isRangeStyleSelection;
```

Comment:
These getters need to be implemented otherwise the compiler will generate storage for these properties redundantly.

Reply:
What do you mean "redundantly"? We want these properties to exist?

Reply:
Yes, but we don't want the compiler to generate ivars for them? If we don't implement the getters we'll have `_isRangeSelection` which is never used.

Reply:
We use those in the constructor 🤔

Reply:
Hmm I suppose I am just confused then 😓